scipy/signal/_peak_finding_utils.pyx

```diff
@@ -210,34 +226,30 @@ def _peak_prominences(np.float64_t[::1] x not None,
                 i_max = min(peak + wlen // 2, i_max)
 
             # Find the left base in interval [i_min, peak]
-            i = peak
+            i = left_bases[peak_nr] = peak
             left_min = x[peak]
             while i_min <= i and x[i] <= x[peak]:
                 if x[i] < left_min:
                     left_min = x[i]
                     left_bases[peak_nr] = i
                 i -= 1
-            if not left_min < x[peak]:
-                raise_error = True  # Raise error outside nogil statement
-                break
 
             # Find the right base in interval [peak, i_max]
-            i = peak
+            i = right_bases[peak_nr] = peak
             right_min = x[peak]
             while i <= i_max and x[i] <= x[peak]:
                 if x[i] < right_min:
                     right_min = x[i]
                     right_bases[peak_nr] = i
                 i += 1
-            if not right_min < x[peak]:
-                raise_error = True  # Raise error outside nogil statement
-                break
 
             prominences[peak_nr] = x[peak] - max(left_min, right_min)
+            if prominences[peak_nr] == 0:
+                show_warning = True
 
-    if raise_error:
-        raise ValueError('{} is not a valid peak'.format(peak))
-
+    if show_warning:
+        warnings.warn("some peaks have a prominence of 0",
+                      PeakPropertyWarning)
```

Comment:
This also needs the `stacklevel` kwarg set in order to get the warning emitted outside `peak_prominences`

Reply:
Oh, cool I didn't know that functionality existed. :)
Is that the best practice in SciPy? Currently the warning points to the internal function `_peak_properties` and is emitted [here](https://github.com/scipy/scipy/blob/ad88cc70e75d0f0539ddf5bccdad4e2e5dd4d71f/scipy/signal/_peak_finding.py#L415). I would have thought that raising the warning closer to the source would  give users and library maintainers an easier time finding the original source of the warning if they want to do something with it.

But I'll add the keyword for now.

Reply:
Yes, the original source in this case is their own code. 
It would not help to debug if their code runs `peak_prominences` and receive a warning inside `_peak_prominences`. Then people would start hunting down what actually caused the emission. 

If stacklevel is set then the warning would be emitted at whatever called `peak_prominences()`. In this case it is `stacklevel=3`

Reply:
Okay, I think I see were you are going with this. However a user might call `peak_prominences` directly (stacklevel should be 2) or implicitly through `find_peaks` (stacklevel should be 3). So which number should I use?

Reply:
Hmm, I could use `_peak_prominences` instead of `peak_prominences` inside `find_peaks`. However I'd have to copy / add some validation for validation `wlen` into `find_peaks`. Then I could use stacklevel=2.

Reply:
Yes please find a way to make the `stacklevel` work for both public functions.